### PR TITLE
#1824 Make `+inf.times` and `-inf.times` work with multiple arguments

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Closes: #1824 

What's done:
- Made `positive-infinity.times` work with multiple arguments
- Made `negative-infinity.times` work with multiple arguments
- Removed caching of `@` in `positive-infinity`, `negative-infinity` and `nan` because it causes `StackOverflow`
- `plus` and `minus` in infinities were refactored. Now they have the same code style
- Added tests
- Added `todo` for future optimization

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@Graur please have a look

File: eo-runtime/src/main/eo/org/eolang/positive-infinity.eo [hunk 1]
```diff
@@ -59,70 +59,145 @@
...
+      try > @
+        []
+          term.gt 0 > @
+        [e]
+          term.gt 0.0 > @
+        FALSE
+
+    # @todo #1824:30min. Pass "is-nan-or-zero" to "times-rec" here and in
+    #  negative-infinity. Now for some reason it's not possible to pass two or
+    #  more objects-functions to another object. To avoid code duplication
```
Comment (Graur) [inline eo-runtime/src/main/eo/org/eolang/positive-infinity.eo:89]: @maxonfjvipon As far as I understand this todo issue, you want to provide to the object more than 3 arguments. If it's it right you can just add all your arguments to new array and pass this array as argument to particular object

**Review by Graur — CHANGES_REQUESTED** [review]
@maxonfjvipon Good one, just one comment from my side

**Comment by maxonfjvipon** [comment]
@Graur could you please review one more time and check what's wrong with windows build

**Comment by maxonfjvipon** [comment]
@Graur please have a look one more time

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**